core/chains/evm/logpoller/orm.go

```diff
@@ -129,3 +129,83 @@ func (o *ORM) SelectLogsByBlockRangeFilter(start, end int64, address common.Addr
 	}
 	return logs, nil
 }
+
+// LatestLogEventSigsAddrsWithConfs finds the latest logs by block that matches a list of addresses and list of events
+func (o *ORM) LatestLogEventSigsAddrsWithConfs(fromBlock int64, addresses []common.Address, eventSigs []common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error) {
+	var logs []Log
+
+	var latestBlocks []int64
+
+	arg := map[string]interface{}{
+		"addresses": addresses,
+		"sigs":      eventSigs,
+		"confs":     confs,
+		"chainid":   utils.NewBig(o.chainID),
+		"fromBlock": fromBlock,
+	}
+
+	// Get the latest block with a matching update for each address
+	query, args, err := sqlx.Named(`
+	SELECT MAX(block_number) as block_number FROM logs
+	WHERE evm_chain_id = :chainid
+	AND address IN (:addresses)
+	AND event_sig IN (:sigs)
+	AND block_number > :fromBlock
+	AND (block_number + :confs) <= (
+		SELECT COALESCE(block_number, 0)
+		FROM log_poller_blocks
+		WHERE evm_chain_id = :chainid
+		ORDER BY block_number DESC
+		LIMIT 1
+	)
+	GROUP BY address, event_sig, evm_chain_id
+	ORDER BY block_number DESC`,
```

Comment:
Can you include the results of running this query with EXPLAIN ANALYZE on a large database with real world data in it?

Reply:
I think this ties in the unpruned logs q as well? cc @connorwstein

Reply:
this is a new query so needs some new benchmarking. You can load up a DB with logs via `TestPopulateLoadedDB` but you'll probably need to modify that to load up log_poller_blocks too

Reply:
I appended to benchmarks and dropped the log_poller_blocks as it seemed very inefficient on larger database (took multiple seconds on a 10k records db)
current benchmark is around 150-200ms

Reply:
benchmark test numbers over 1m record
```
/usr/bin/go test -v -timeout 30s -run ^TestPopulateLoadedDB github.com/smartcontractkit/chainlink/core/chains/evm/logpoller -ldflags="-X github.com/smartcontractkit/chainlink/core/static.Version=1.1.0 -X github.com/smartcontractkit/chainlink/core/static.Sha=174e80cb4b763e8f422282580e392a60f10fa237 -X github.com/smartcontractkit/chainlink/core/static.SentryDSN="
=== RUN   TestPopulateLoadedDB
    integration_test.go:53: 96.360329ms 25000 // <- this call to SelectLogsByBlockRangeFilter
    integration_test.go:59: 162.912335ms 1 // <- this is our query call LatestLogEventSigsAddrs
--- PASS: TestPopulateLoadedDB (17.96s)
PASS
ok  	github.com/smartcontractkit/chainlink/core/chains/evm/logpoller	18.018s
```